api/middleware/stats.go

```diff
@@ -0,0 +1,86 @@
+package middleware
+
+import (
+	"fmt"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/raintank/metrictank/stats"
+	"gopkg.in/macaron.v1"
+)
+
+type requestStats struct {
+	sync.Mutex
+	responseCounts    map[string]map[int]*stats.Counter32
+	latencyHistograms map[string]*stats.LatencyHistogram15s32
+	sizeMeters        map[string]*stats.Meter32
+}
+
+func (r *requestStats) PathStatusCount(path string, status int) {
+	r.Lock()
+	p, ok := r.responseCounts[path]
+	if !ok {
+		p = make(map[int]*stats.Counter32)
+		r.responseCounts[path] = p
+	}
+	c, ok := p[status]
+	if !ok {
+		c = stats.NewCounter32(fmt.Sprintf("api.request.%s.%d", pathSlug(path), status))
```

Comment:
I think the key should be `api.request.<path>.status.<statuscode>` that way in the tree under path we have size, latency and status, instead of latency, size , 400, 200, 503, etc which aren't concepts that go together in the same level.